mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -105,6 +131,10 @@ where
 
         let result_coin = (self.constructor)(utxo_arc.clone());
         if let Some(sync_status_loop_handle) = sync_status_loop_handle {
+            // Try adding the given starting block to storage if it's needed to. SPVConf::start_block_header and
+            // verification_params must be set!
+            add_start_block_header_to_storage(result_coin.clone()).await?;
```

Comment:
I don't think this is the best place to add the starting block header to storage in, a better place can be here https://github.com/KomodoPlatform/atomicDEX-API/blob/526ef03bf6289e6a92f7a034c241e01f03784256/mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs#L296

Reply:
https://github.com/KomodoPlatform/atomicDEX-API/blob/d55826889f9ec8bf66377c73c6139c840ad30462/mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs#L275
If it's must be set then it should be non-optional. Before this PR it was non-optional too since the genesis header was required.
Please consider the following changes https://github.com/KomodoPlatform/atomicDEX-API/compare/spv-optimizations...KomodoPlatform:atomicDEX-API:spv-starting-header

Reply:
ok thank you!